gamemode/modules/inventory/cl_inventory.lua

```diff
@@ -0,0 +1,322 @@
+        -- Open own inventory
+        net.Start( "RVR_Inventory_Open" )
+        net.SendToServer()
+
+        return true
+    elseif string.StartWith( bind, "slot" ) then
+        -- Set selected slot
+        local slotNum = tonumber( string.sub( bind, 5 ) )
+        if not slotNum then return end
+
+        inv.setHotbarSlot( slotNum )
+
+        return true
+    end
+end )
+
+function inv.closeInventory()
+    if inv.openInventory then
+        inv.openInventory:Remove()
+        inv.openInventory = nil
+    end
+
+    inv.disableCursorSlot()
+    net.Start( "RVR_Inventory_Close" )
+    net.SendToServer()
+end
+
+function inv.enableCursorSlot()
+    inv.showCursorItem = true
+    inv.clearCursorItemData()
+end
+
+function inv.disableCursorSlot()
+    inv.showCursorItem = false
+    inv.clearCursorItemData()
+end
+
+function inv.setCursorItemData( item, count )
+    inv.cursorItem = item
+    inv.cursorItemMaterial = Material( string.sub( item.icon, 11 ) ) -- Remove "material/" from icon path
+    inv.cursorItemCount = count
+end
+
+function inv.clearCursorItemData()
+    inv.cursorItem = nil
+    inv.cursorItemMaterial = nil
+    inv.cursorItemCount = nil
+end
+
+function inv.getCursorItemData()
+    return inv.cursorItem, inv.cursorItemCount
+end
+
+-- Render the cursor slot
+hook.Add( "PostRenderVGUI", "RVR_Inventory_DrawCursorItem", function()
+    if not inv.showCursorItem or not inv.cursorItemMaterial then return end
+
+    local mx, my = gui.MousePos()
+    local x, y = mx - cursorSlotSize / 2, my - cursorSlotSize / 2
+
+    -- Item icon
+    surface.SetDrawColor( 255, 255, 255, 255 )
+    surface.SetMaterial( inv.cursorItemMaterial )
+    surface.DrawTexturedRect( x, y, cursorSlotSize, cursorSlotSize )
+
+    if inv.cursorItemCount <= 1 then return end
+
+    -- Item count
+    local countText = tostring( inv.cursorItemCount )
+    surface.SetFont( "DermaLarge" )
+    local tw, th = surface.GetTextSize( countText )
+    local tx = x + cursorSlotSize - tw - 1
+    local ty = y + cursorSlotSize - th + 5
+
+    surface.SetTextColor( 200, 200, 200 )
+    surface.SetTextPos( tx, ty )
+    surface.DrawText( countText )
+end )
+
+-- Handle dropping items by clickout outside of openInventory frame
+hook.Add( "GUIMousePressed", "RVR_Inventory_DropItem", function( code, aimVector )
+    if not inv.openInventory then return end
+    local mx, my = gui.MousePos()
+
+    local panelX, panelY = inv.openInventory:GetPos()
+    local panelW, panelH = inv.openInventory:GetSize()
+
+    local onBorderX = mx < panelX or mx > panelX + panelW
+    local onBorderY = my < panelY or my > panelY + panelH
+
+    if not onBorderX and not onBorderY then return end
+
+    local item, count = inv.getCursorItemData()
+    -- Don't drop if no item
+    if not item then return end
+
+    -- Right clicking only drops one item
+    if code == MOUSE_RIGHT then count = 1 end
+
+    net.Start( "RVR_Inventory_CursorDrop" )
+    net.WriteInt( count, 8 )
```

Comment:
You could change that to net.WriteUInt() as I am pretty sure you'll never drop a negative amount of items. That would allow you dropping 255 items instead of 127 for the same bit amount. You could also send only 4 bits but at this point I don't think it makes much of a difference

Reply:
Solid point, theres probably a few places UInt's would be better, i'll look into it :)